backend/wmg/data/load_cube.py

```diff
@@ -13,37 +13,186 @@
 
 stack_name = os.environ.get("REMOTE_DEV_PREFIX")
 wmg_bucket_name = os.environ.get("WMG_BUCKET")
-artifact_bucket_name = os.environ.get("ARTIFACT_BUCKET")
 
+# TODO(prathap): The functionality to fully integrate `snapshot_schema_version` will be done
+# in a future ticket:
+# https://github.com/chanzuckerberg/single-cell-data-portal/issues/5166
+
+###################################### PUBLIC FUNCTIONS #################################
+
+
+@log_func_runtime
+def upload_artifacts_to_s3(
+    *, snapshot_source_path: str, snapshot_schema_version: str, snapshot_id: int, is_snapshot_validation_successful=True
+) -> str:
+    """
+    Uploads the generated cubes to S3 and writes metadata that includes
+    information about the data schema version, the snapshot id, and the location
+    of the snapshot.
+
+    TODO(prathap): This part of the docstring will be modified when work starts on ticket:
+    https://github.com/chanzuckerberg/single-cell-data-portal/issues/5166
+
+    Specifically:
+
+    1. Snapshot artifacts are stored in s3://<environment_bucket_name>/<snapshot_id>/
+       A. Note that, as of now, <snapshot_id> is a integer representing the time elapsed
+          in seconds since epoch
+       B. If the snapshot generation pipeline is run in a "remote dev environment", then the snapshot
+          is stored in s3://<environment_bucket_name>/<remote_stack_name>/<snapshot_id>/
+       C. If validation checks on the snapshot fails, the data artifact will be stored in
+          s3://<environment_bucket_name>/<snapshot_id>_validation_failed_snapshot/
+          (or s3://<environment_bucket_name>/<remote_stack_name>/<snapshot_id>_validation_failed_snapshot/)
```

Comment:
Oh the failed snapshot going to be originally uploaded with the "validation_failed" infix, right? As opposed to renaming an S3 "dir".

Reply:
@atolopko-czi - That's right. Moreover, this PR doesn't change any existing behavior (_yet_), except for adding another file called `latest_snapshot_run`. This `latest_snapshot_run` file won't be used by the reader but more meant as an easy way for operators to check where the latest run is located given that the new design will store snapshots in `s3://<bucket_name/snapshots/<data_schema_version>/` directory and it would be useful to have a `s3://<bucket_name>/latest_snapshot_run` at the top level that contains the path to the latest snapshot run (ex: `s3://<bucket_name/snapshots/v2/1688370972`)